core/src/main/java/com/linecorp/armeria/server/http/dynamic/Methods.java

```diff
@@ -0,0 +1,126 @@
+     *   <li> {@link Trace} -> {@link HttpMethod#TRACE}
+     *   <li> {@link Connect} -> {@link HttpMethod#CONNECT}
+     * </ul>
+     */
+    private static final Map<Class<?>, HttpMethod> HTTP_METHOD_MAP =
+            ImmutableMap.<Class<?>, HttpMethod>builder()
+                    .put(Options.class, HttpMethod.OPTIONS)
+                    .put(Get.class, HttpMethod.GET)
+                    .put(Head.class, HttpMethod.HEAD)
+                    .put(Post.class, HttpMethod.POST)
+                    .put(Put.class, HttpMethod.PUT)
+                    .put(Patch.class, HttpMethod.PATCH)
+                    .put(Delete.class, HttpMethod.DELETE)
+                    .put(Trace.class, HttpMethod.TRACE)
+                    .put(Connect.class, HttpMethod.CONNECT)
+                    .build();
+
+    /**
+     * Returns the array of {@link Path} annotated methods.
+     */
+    public static Method[] requestMappingMethods(Object object) {
+        List<Method> methods = Arrays.stream(object.getClass().getMethods())
+                                     .filter(m -> null != m.getAnnotation(Path.class))
+                                     .collect(Collectors.toList());
+        return methods.toArray(new Method[methods.size()]);
+    }
+
+    /**
+     * Returns the array of parameter types.
+     */
+    public static Class<?>[] parameterTypes(Method method) {
+        int parameterCount = method.getParameterCount();
+        return Arrays.stream(method.getParameters())
+                     .map(p -> p.getType())
+                     .toArray(size -> new Class<?>[parameterCount]);
+    }
+
+    /**
+     * Returns the array of path variable values of parameters.
+     */
+    public static String[] pathVariables(Method method) {
+        int parameterCount = method.getParameterCount();
+        return Arrays.stream(method.getParameters())
+                     .map(p -> p.getAnnotation(PathParam.class))
+                     .map(a -> a.value())
+                     .toArray(size -> new String[parameterCount]);
+    }
+
+    /**
+     * Returns {@link EnumSet} instance of {@link HttpMethod} mapped to {@code method}. If no specific HTTP
+     * Method is mapped to given {@code method}, it is regarded as all HTTP Methods are mapped to on it.
+     *
+     * @see Options
+     * @see Get
+     * @see Head
+     * @see Post
+     * @see Put
+     * @see Patch
+     * @see Delete
+     * @see Trace
+     * @see Connect
+     */
+    public static EnumSet<HttpMethod> httpMethods(Method method) {
+        EnumSet<HttpMethod> ret = EnumSet.noneOf(HttpMethod.class);
+        for (Annotation annotation : method.getAnnotations()) {
+            HttpMethod mappedMethod = HTTP_METHOD_MAP.get(annotation.annotationType());
+            if (mappedMethod != null) {
+                ret.add(mappedMethod);
+            }
+        }
+
+        if (ret.isEmpty()) {
+            ret = EnumSet.allOf(HttpMethod.class);
```

Comment:
No problem. Applying this, I added some validation routine for checking @Path annotated method without any HTTP Method annitations, e.g., @Get, @Post, etc in DynamicHttpServiceBuilder#add(Object).